### PR TITLE
MESH-1870/ During ticker registration reject ASCII control characters (non printable)

### PR DESCRIPTION
**Pull request by JMoore96**
### modified logic

- created a check in ticker registration to make sure it is alphanumeric

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -1200,13 +1201,16 @@ impl<T: Config> Module<T> {
     /// Ensure `ticker` is fully printable ASCII (SPACE to '~').
     fn ensure_ticker_ascii(ticker: &Ticker) -> DispatchResult {
         let bytes = ticker.as_slice();
+        let s = String::from_utf8_lossy(bytes);
+        let result = s.chars().all(char::is_alphanumeric);
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: This check will reject trailing NULL bytes, which we want to allow.  The alphanumeric check should be done in the look below.

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -1200,13 +1201,16 @@ impl<T: Config> Module<T> {
...
         // Find first byte not printable ASCII.
         let good = bytes
             .iter()
             .position(|b| !matches!(b, 32..=126))
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: This line should be changed to find the first non-alphanumeric charater.

File: pallets/asset/src/lib.rs [hunk 3]
```diff
@@ -1207,6 +1208,11 @@ impl<T: Config> Module<T> {
             // Everything after must be a NULL byte.
             .map_or(true, |nm_pos| bytes[nm_pos..].iter().all(|b| *b == 0));
         ensure!(good, Error::<T>::TickerNotAscii);
+
+        for ref_b in bytes {
+            let deref_b = *ref_b;
+            ensure!(deref_b.is_ascii_alphanumeric(), Error::<T>::TickerNotAscii);
+        }
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: This check will not allow trailing NULL bytes.

This code on line 1206, needs to be changed to use the `is_ascii_alphanumeric` call.
```
.position(|b| !matches!(b, 32..=126))
```

**Review by Neopallium — CHANGES_REQUESTED** [review]

File: pallets/asset/src/lib.rs [hunk 4]
```diff
@@ -1203,7 +1203,10 @@ impl<T: Config> Module<T> {
         // Find first byte not printable ASCII.
         let good = bytes
             .iter()
-            .position(|b| !matches!(b, 32..=126))
+            .position(|b| {
+                let deref_b = *b;
+                !deref_b.is_ascii_alphanumeric()
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: Parentheses can be used to avoid the local variable.
```suggestion
                !(*b).is_ascii_alphanumeric()
```

**Comment by Neopallium** [comment]
Looks like some of the reserved classic tickers don't meet the new rules.

**Comment by Neopallium** [comment]
- [x] Clear out classic tickers from the `src/data/reserved_classic_tickers.json`.  Just keep about 2 tickers from that file.

**Review by Neopallium — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**